### ISSUE TITLE
The results generated by GPU and CPU are different when running example LLM program

### ISSUE THREAD

**Issue by fenghao999** [intel/ipex-llm#13140]
When I run the same LLM model over IPEX-LLM using the same example program, the generated context is different between running on Intel GPU and running on Intel CPU given certain input prompt, even though the program is the same, and “do_sample=False” is set. I think they should generate exactly the same context. I don't understand why this happens. 

I ran the example program for **Llama 2-13b-chat-hf** model with "**sym_int8**" quantization and input the prompt "**What is AI?**". The other specific settings are as follows:
1. The example program I ran is ".../ipex-llm/python/llm/example/GPU/HuggingFace/LLM/llama2/generate.py". When running on GPU, I moved the model and input_ids to GPU by using function .to("xpu"), while moving the mode and input_ids to CPU by updating the function .to("cpu") when running on CPU.
2. I am using Linux system equipped with Intel Max GPU 1550 and Intel CPU Xeon Platinum 8468V (4th generation). 
3. I upgraded related libraries’ versions by doing: pip install transformers==4.45.0  / pip install accelerate==0.33.0 / pip install "trl<0.12.0", which should be the same for GPU and CPU.

Again, although I have set “do_sample=False” as greedy selection of the next token based on calculated logits, the generated context is different between running GPU and CPU.

Running on Intel Max GPU:
![Image](https://github.com/user-attachments/assets/ff5b7509-a4f1-44b7-8bcf-75a9f89c2fa6)

Running on Intel CPU:
![Image](https://github.com/user-attachments/assets/708b3aef-6287-4021-8bfd-3ba344c59d76)

I believe CPU and GPU should generate the same context. I tried a couple of other input prompt, sometime CPU and GPU generate the same context. However, with some prompts, such as "What is AI?", they generate different context.

**Comment by hkvision** [comment]
Hi,

For CPU and GPU, there are different optimizations in our implementation, and thus would cause slight difference in output logits and thus the different output in some cases is expected.
